src/workerd/util/sqlite.c++

```diff
@@ -640,6 +643,13 @@ SqliteDatabase::StatementAndEffect SqliteDatabase::prepareSql(
 
           // Reduce `sqlCode` to include only what we haven't already executed.
           sqlCode = kj::StringPtr(tail, sqlCode.end());
+
+          KJ_IF_SOME(p, prelude) {
+            p.add(Statement(*this, regulator,
+                StatementAndEffect{.statement = kj::mv(ownResult),
+                  .stateChange = kj::mv(parseContext.stateChange)}));
```

Comment:
Eek, no, that is a bug. Good catch. I'd better add a test that does transactions.

(Though this bug wouldn't be possible to hit from JavaScript since we don't allow people to execute transaction statements from JS.)

Reply:
(But it's not actually an uninialized OneOf. The default value for `ParseContext.stateChange` is `NoChange()`.)